compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5188,110 +5316,43 @@ private boolean isFunctionPointer(BSymbol funcSymbol) {
                 && (funcSymbol.flags & Flags.NATIVE) != Flags.NATIVE;
     }
 
-    private void reportErrorDetailMissmatchError(BLangErrorConstructorExpr errorConstructorExpr,
-                                                 BRecordType targetErrorDetailRec,
-                                                 BRecordType recordType) {
-        boolean detailedErrorReported = false;
-        Set<String> checkedFieldNames = new HashSet<>();
-        for (Map.Entry<String, BField> fieldEntry : targetErrorDetailRec.fields.entrySet()) {
-            checkedFieldNames.add(fieldEntry.getKey());
-            BField argField = recordType.fields.get(fieldEntry.getKey());
-            if (argField == null && !Symbols.isOptional(fieldEntry.getValue().symbol)) {
-                dlog.error(errorConstructorExpr.pos, DiagnosticErrorCode.MISSING_ERROR_DETAIL_ARG, fieldEntry.getKey());
-                detailedErrorReported = true;
-            } else if (!types.isAssignable(argField.type, fieldEntry.getValue().type)) {
-                dlog.error(errorConstructorExpr.pos, DiagnosticErrorCode.INVALID_ERROR_DETAIL_ARG_TYPE,
-                        fieldEntry.getKey(), fieldEntry.getValue().type, argField.type);
-            }
-        }
+    // todo: try to re-use recrod literal checking
```

Comment:
Are we planning on introducing any changes for this?

Reply:
No will remove, this PR is already re-using mapping constructor check